public/docs/ts/latest/cookbook/a11y.jade

```diff
@@ -0,0 +1,1087 @@
++makeExample('cb-a11y/ts/app/managing-focus/a11y-managing-focus.component.html','cb-a11y-managing-focus-skiplinks-destination')
+
+:marked
+  Finally we need some styling magic to make it all work by only showing focused links:
+
++makeExample('cb-a11y/ts/a11y.css', 'cb-a11y-managing-focus-skiplinks-style')
+
+:marked
+  Now that we know more about how browsers handle focus and what we can do to leverage it, it is time to look at what
+  we can do inside our custom Angular&nbsp;2 components to keep them accessible.
+
+:marked
+  ### Interactive components should accept focus
+  
+  Unlike native interactive HTML elements, `custom interactive components` created with Angular&nbsp;2
+  won't accept focus within the normal focus flow of the page during `keyboard navigation` unless
+  we plan for it.
+  
+  This means that a native `button` element will accept focus, but a `button control` built as a 
+  `custom element` from non-interactive HTML elements won't.
+  
+  So let's do exactly that, but first, another word of warning.
+
+.l-sub-section
+  :marked
+    Again we need to stress that re-creating any `native HTML element` out of `custom elements` is **NOT** recommended, nor
+    do we make any promises about the production readiness of our example. Our focus remains on accessibility and how
+    we have the tools to make the most stubbornly inaccessible component accessible.
+
+:marked
+  There are rules governing which keyboard events should ideally be implemented per widget.
+  For example, a regardless of how it is accessed, a `button` should accept focus, react to the
+  mouse `click` event and react to the keyboard `enter` and `space` events.
+
+  *You can read about these [Common Widget Design Patterns](https://www.w3
+  .org/WAI/intro/accessibility.php) at the `W3C`.*
+
+  Let's take a look at our custom button element:
+
++makeTabs('cb-a11y/ts/app/shared/a11y-custom-button.component.ts,cb-a11y/ts/app/shared/a11y-custom-button.component.html',
+null, 'a11y-custom-button.component.ts,a11y-custom-button.component.html')
+
+:marked
+  We manipulate the `Host` element of our component and now it can now be used like the standard
+  `button` element:
+
++makeExample('cb-a11y/ts/app/managing-focus/a11y-managing-focus.component.html','cb-a11y-custom-button-usage')
+
+:marked
+  Looking at the generated HTML we see:
+
+code-example(language="html" escape="html" format="linenums").
+  <a11y-custom-button class="btn btn-primary" role="button" tabindex="0">
+    Do something...
+  </a11y-custom-button>
+
+:marked
+  **Important**:  Note the `role` and `tabindex`. 
+  
+  The `ARIA role` of this element is `button`. It tells any assistive technologies that this element is
+  a button, regardless of the original design of the HTML element. More information on this later.
+  
+  Setting the `tabindex` to `0` inserts the element in the default flow of`keyboard navigation` focus. This means that our
+  element also becomes accessible via the keyboard!
```

Comment:
IMO this is a little misleading since it's really the combination of `tabindex` and the `keydown` bindings that make it accessible via keyboard. So I'm wondering if it'd be more effective to just walk through the whole implementation of the component vs. calling out these two items specifically.